### PR TITLE
Improve tests webvttfile accordingly to the web video text tracks format specification

### PR DESCRIPTION
**Pull request by RamanYeda**
Hello. I tryed to optimize a parse method mainly. Also there was added vtt files from https://github.com/w3c/webvtt#webvtt specification and written a unit tests for `WebvttFile` class.

### CODE REVIEW

File: src/Captioning/Format/SubripCue.php [hunk 1]
```diff
@@ -8,8 +8,9 @@ class SubripCue extends Cue
 {
     public static function tc2ms($tc)
     {
-        $tab = explode(':', $tc);
-        $durMS = $tab[0] * 60 * 60 * 1000 + $tab[1] * 60 * 1000 + floatval(str_replace(',', '.', $tab[2])) * 1000;
+        $tab = array_reverse(explode(':', $tc));
+        $tab[2] = isset($tab[2]) ? $tab[2] : 0;
+        $durMS = $tab[2] * 60 * 60 * 1000 + $tab[1] * 60 * 1000 + floatval(str_replace(',', '.', $tab[0])) * 1000;
```
Comment (RamanYeda) [inline src/Captioning/Format/SubripCue.php]: There is valid case in a exapmle https://w3c.github.io/webvtt/#introduction. Previous code will fail when trying to parse vtt from there.

File: src/Captioning/Format/WebvttFile.php [hunk 2]
```diff
@@ -6,98 +6,92 @@
 
 class WebvttFile extends File
 {
-    const TIMECODE_PATTERN = '#^([0-9]{2}:[0-9]{2}:[0-9]{2}.[0-9]{3}) --> ([0-9]{2}:[0-9]{2}:[0-9]{2}.[0-9]{3}) ?(.*)$#';
+    const TIMECODE_PATTERN = '#^((?:[0-9]{2}:)?[0-9]{2}:[0-9]{2}.[0-9]{3}) --> ((?:[0-9]{2}:)?[0-9]{2}:[0-9]{2}.[0-9]{3})( .*)?$#';
```
Comment (RamanYeda) [inline src/Captioning/Format/WebvttFile.php]: Corrected to right match patterns like in https://w3c.github.io/webvtt/#introduction example.

File: src/Captioning/Format/WebvttFile.php [hunk 3]
```diff
@@ -6,98 +6,92 @@
...
 
     protected $regions = array();
 
     public function parse()
```
Comment (RamanYeda) [inline src/Captioning/Format/WebvttFile.php]: Parsing algorithm is completely rewritten and now it has a more rational form (in my opinion).

File: tests/Captioning/Format/WebvttFileTest.php [hunk 4]
```diff
@@ -4,6 +4,146 @@
 
 class WebvttFileTest extends \PHPUnit_Framework_TestCase
```
Comment (RamanYeda) [inline tests/Captioning/Format/WebvttFileTest.php]: Additional unit tests. I get some examples from specification and test parse method. After that I also tested exeptions what parse method can throws throuh parsing process.

**Comment by RamanYeda** [comment]
Wait for your opinion, @delphiki

**Comment by delphiki** [comment]
Wow, thanks a lot for this PR.
Sorry for coming back at you more than a month later, had a really busy weeks lately (+ the lack of motivation I have to admit).

Seems like a nice update of the webvtt parser.
I'll check this at home this week-end (currently writing those lines on a train, not the best conditions to review code).

**Comment by RamanYeda** [comment]
Ok, I will be waiting. Thanks.

File: src/Captioning/Format/SubripCue.php (hunk 1 above)
Comment (delphiki) [inline src/Captioning/Format/SubripCue.php]: Nicely done!

File: src/Captioning/Format/WebvttFile.php [hunk 5]
```diff
@@ -6,98 +6,92 @@
...
+                    ++$i;
+                }
+
+                // Make the cue object and add it to the file.
+                $cue = $this->createCue($start, $stop, $text, $settings, $id, $note);
+                $note = $id = '';
+                $this->addCue($cue);
+                unset($cue);
+            } elseif ($line !== '') {
+                // Supposse what not empty line before timeline is id.
```
Comment (delphiki) [inline src/Captioning/Format/WebvttFile.php]: suppose* :)

**Comment by delphiki** [comment]
As I thought, great work, the parser has now a way better structure.

And great set of tests!

Thanks a lot!

File: src/Captioning/Format/WebvttFile.php (hunk 5 above)
Comment (RamanYeda) [inline src/Captioning/Format/WebvttFile.php]: oops, sorry )

**Comment by hartman** [comment]
:tada:

**Pull request merged**